clients/fides-js/src/components/tcf/DoubleToggleTable.tsx

```diff
@@ -0,0 +1,166 @@
+  items: T[];
+  title: string;
+  enabledConsentIds: string[];
+  enabledLegintIds: string[];
+  renderToggleChild: (item: T) => VNode;
+  onToggle: (payload: UpdateEnabledIds) => void;
+  renderBadgeLabel?: (item: T) => string | undefined;
+  consentModelType: keyof EnabledIds;
+  legintModelType: keyof EnabledIds;
+}
+
+const DoubleToggleTable = <T extends Item>({
+  items,
+  title,
+  enabledConsentIds,
+  enabledLegintIds,
+  renderToggleChild,
+  renderBadgeLabel,
+  onToggle,
+  consentModelType,
+  legintModelType,
+}: Props<T>) => {
+  const toggleAllId = `all-${title}`;
+  const toggleAllConsentId = `${toggleAllId}-consent`;
+
+  const { allConsentChecked, allLegintChecked } = useMemo(() => {
+    const consentIds = items.filter((i) => i.isConsent).map((i) => `${i.id}`);
+    const legintIds = items.filter((i) => i.isLegint).map((i) => `${i.id}`);
+
+    const consentChecked = consentIds.every(
+      (id) => enabledConsentIds.indexOf(id) !== -1
+    );
+    const legintChecked = legintIds.every(
+      (id) => enabledLegintIds.indexOf(id) !== -1
+    );
+    return {
+      allConsentChecked: consentChecked,
+      allLegintChecked: legintChecked,
+    };
+  }, [items, enabledConsentIds, enabledLegintIds]);
+
+  const { hasConsentItems, hasLegintItems } = useMemo(
+    () => ({
+      hasConsentItems: !!items.filter((i) => i.isConsent).length,
+      hasLegintItems: !!items.filter((i) => i.isLegint).length,
+    }),
+    [items]
+  );
+
+  const handleToggleAll = (legalBasis: LegalBasisEnum) => {
+    const isConsent = legalBasis === LegalBasisEnum.CONSENT;
+    const modelType = isConsent ? consentModelType : legintModelType;
+    const allChecked = isConsent ? allConsentChecked : allLegintChecked;
+    if (allChecked) {
+      onToggle({ newEnabledIds: [], modelType });
+    } else {
+      const allItems = isConsent
+        ? items.filter((i) => i.isConsent)
+        : items.filter((i) => i.isLegint);
+      onToggle({
+        newEnabledIds: allItems.map((i) => `${i.id}`),
+        modelType,
+      });
+    }
+  };
+
+  const handleToggleIndividual = (item: T) => {
+    const enabledIds = item.isConsent ? enabledConsentIds : enabledLegintIds;
+    const modelType = item.isConsent ? consentModelType : legintModelType;
+    const purposeId = `${item.id}`;
+    if (enabledIds.indexOf(purposeId) !== -1) {
+      onToggle({
+        newEnabledIds: enabledIds.filter((e) => e !== purposeId),
+        modelType,
+      });
+    } else {
+      onToggle({
+        newEnabledIds: [...enabledIds, purposeId],
+        modelType,
+      });
+    }
+  };
+
+  return (
+    <div>
+      {/* DEFER: ideally we use a table object, but then DataUseToggles would need to be reworked
+      or we would need a separate component. */}
+      <div className="fides-legal-basis-labels">
+        <span className="fides-margin-right">Legitimate interest</span>
+        <span>Consent</span>
+      </div>
+      <DataUseToggle
```

Comment:
the one thing that did get added which `TcfVendors` didn't have is this title row which also has toggles:

![image](https://github.com/ethyca/fides/assets/24641006/0bf31e62-0e27-4270-b6e2-8bed219152ce)